### PR TITLE
Add node-level task transforms

### PR DESCRIPTION
**Pull request by JSSchmidt**
Adds `AttributeToNodeRegressionTransform(source)` and `AttributeToNodeClassTransform(source, mapping)` in `mantra/transforms/node_task_transforms.py`.

Both turn an attribute holding one value per vertex into `data.y`: a float `(n_vertices, 1)` target for regression, or long `(n_vertices,)` class indices from a fixed `mapping` for classification, one entry per vertex in vertex order. Every vertex is a target; there is no mask, so a consumer that wants to leave a vertex out does so itself.

Like the graph-level task transforms from #94 they are stateless and meant to run as `transform=`. In line with `AttributeToRegressionTransform`, the attribute is read with `torch.as_tensor`, so lists (`pre_transform` path) and tensors (`transform` path) both work; the only shape requirement is a 1-D vector. The class variant rejects float values and raises the same informative `KeyError` as `AttributeToClassTransform` for values outside the mapping.

**Tests:** `test/test_node_task_transforms.py` (shapes/dtypes, list and tensor inputs, batching, validation errors). Two tests pin the node alignment: row *k* of `y` belongs to node *k* (the node whose local feature is *k*), and after `Batch.from_data_list` the rows follow `batch.batch` / `ptr` and edge endpoints (offset by the batching) index the right targets. README and CHANGELOG updated.

Targets `cy`, which already contains `main`. Consumed by the node-level Calabi-Yau task in mantra-pp-benchmarks.

### CODE REVIEW

File: mantra/transforms/node_task_transforms.py
Comment (martin-carrasco) [inline mantra/transforms/node_task_transforms.py:1]: I think this should go in the task transform instead of another file directly

**Review by martin-carrasco — CHANGES_REQUESTED** [review]

File: mantra/transforms/node_task_transforms.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        return len(self.mapping)
+
+    def forward(self, data: Data):
+        values = self._node_values(data)
+
+        data.y = torch.tensor(
+            [self.mapping[value] for value in values.tolist()],
+            dtype=torch.long,
+        )
+        data.node_mask = self._node_mask(values.numel())
```
Comment (martin-carrasco) [inline mantra/transforms/node_task_transforms.py]: The node mask is still here, do we need it and why ?

**Review by martin-carrasco — CHANGES_REQUESTED** [review]

**Review by martin-carrasco — APPROVED** [review]
